pilot/pkg/bootstrap/server.go

```diff
@@ -1243,14 +1245,66 @@ func (s *Server) initWorkloadTrustBundle() {
 	})
 
 	// IstioCA: Explicitly add roots corresponding to CA
-	if s.CA != nil {
+	rootCertFile := path.Join(LocalCertDir.Get(), "root-cert.pem")
+	if _, err := os.Stat(rootCertFile); err != nil {
+		rootCertExists = false
+	}
+	if rootCertExists {
```

Comment:
do we already monitor it in the CA code? since if we update here but not in the signing code, we will have issues?

Reply:
we don't monitor in CA code for now. Agreed, these will need to change together.

Reply:
Have removed this functionality in this PR. Have added functionality to explicitly check the CA certs (all kinds) only on startup. Will add this functionality (dynamically checking trustAnchor) as part of a future PR.